lib/device_detector/version_extractor.rb

```diff
@@ -18,13 +18,17 @@ def extract_version
     end
 
     def replace_version_string_with(match_data)
-      string = version_string
+      string = version_string.to_s
```

Comment:
Shouldn't this be better done in the `#version_string` method?

You could use the following pattern:

```ruby
def version_string
  String(regex_meta['version'])
end
```